pkg/cache/snapshot.go

```diff
@@ -157,26 +159,41 @@ func (c *Cache) Snapshot(ctx context.Context) (*Snapshot, error) {
 
 // snapshotClusterQueue creates a copy of ClusterQueue that includes
 // references to immutable objects and deep copies of changing ones.
-func snapshotClusterQueue(c *clusterQueue) *ClusterQueueSnapshot {
+func (c *Cache) snapshotClusterQueue(ctx context.Context, cq *clusterQueue) *ClusterQueueSnapshot {
+	log := ctrl.LoggerFrom(ctx)
 	cc := &ClusterQueueSnapshot{
-		Name:                          c.Name,
-		ResourceGroups:                make([]ResourceGroup, len(c.ResourceGroups)),
-		FlavorFungibility:             c.FlavorFungibility,
-		FairWeight:                    c.FairWeight,
-		AllocatableResourceGeneration: c.AllocatableResourceGeneration,
-		Workloads:                     maps.Clone(c.Workloads),
-		Preemption:                    c.Preemption,
-		NamespaceSelector:             c.NamespaceSelector,
-		Status:                        c.Status,
-		AdmissionChecks:               utilmaps.DeepCopySets(c.AdmissionChecks),
-		ResourceNode:                  c.resourceNode.Clone(),
+		Name:                          cq.Name,
+		ResourceGroups:                make([]ResourceGroup, len(cq.ResourceGroups)),
+		FlavorFungibility:             cq.FlavorFungibility,
+		FairWeight:                    cq.FairWeight,
+		AllocatableResourceGeneration: cq.AllocatableResourceGeneration,
+		Workloads:                     maps.Clone(cq.Workloads),
+		Preemption:                    cq.Preemption,
+		NamespaceSelector:             cq.NamespaceSelector,
+		Status:                        cq.Status,
+		AdmissionChecks:               utilmaps.DeepCopySets(cq.AdmissionChecks),
+		ResourceNode:                  cq.resourceNode.Clone(),
 		TASFlavors:                    make(map[kueue.ResourceFlavorReference]*TASFlavorSnapshot),
-		tasOnly:                       c.isTASOnly(),
-		flavorsForProvReqACs:          c.flavorsWithProvReqAdmissionCheck(),
+		tasOnly:                       cq.isTASOnly(),
+		flavorsForProvReqACs:          cq.flavorsWithProvReqAdmissionCheck(),
 	}
-	for i, rg := range c.ResourceGroups {
+	for i, rg := range cq.ResourceGroups {
 		cc.ResourceGroups[i] = rg.Clone()
 	}
+	if cq.AdmissionScope != nil {
+		cc.AdmissionScope = *cq.AdmissionScope.DeepCopy()
+	}
+	afsEnabled, resourceWeights := queue.AfsResourceWeights(&cc.AdmissionScope, c.admissionFairSharing)
+	if afsEnabled {
+		for _, wl := range cc.Workloads {
+			usage, err := wl.GetLocalQueueUsage(ctx, c.client, resourceWeights)
+			if err != nil {
+				log.V(2).Error(err, "Error determining LocalQueue usage")
```

Comment:
Silencing errors is typically a code smell, can we propage the error up?

Reply:
This function is called by a scheduler where so far we only log errors and continue with another Workload. Is there anything you'd suggest doing instead? Also the `wl.LocalQueueUsage` is a pointer and later on we check if this is not nil

Reply:
Done